1119 - Add Samples to Dataset Modal

avrohomgottlieb opened the pull request:
## Issue Number

Closes https://github.com/AlexsLemonade/scpca-portal/issues/1119.

## Purpose/Implementation Notes

<!-- For changes that impact the frontend and user interactions, please ensure there's a vercel preview and tag a designer (@dvenprasad) for review  -->

This PR introduces the `DatasetAddSamplesModal` component, which will be triggered following users adding samples to their cart. Only the rendered modal is included in this PR, the functionality will be added over time.

The latest storybook deployment for this component can be found [here](https://scpca-portal-39bk66jk6-ccdl.vercel.app/?path=/story/components-datasetaddsamplesmodal--default).

## Types of changes

What types of changes does your code introduce?

<!-- Remove any which your PR isn't -->


- New feature (non-breaking change which adds functionality)

## Functional tests

N/A

## Checklist

<!-- Put an `x` in the boxes that apply. -->

- [x] Lint and unit tests pass locally with my changes

## Screenshots

N/A

davidsmejia commented on client/src/components/DatasetAddSamplesModal.js:
```diff
@@ -0,0 +1,84 @@
+import { DatasetProjectDataFormat } from 'components/DatasetProjectDataFormat'
+import { DatasetSampesProjectOptions } from 'components/DatasetSamplesProjectOptions'
+
+const Li = ({ text }) => (
+  <Box as="li" style={{ display: 'list-item' }}>
+    {text}
+  </Box>
+)
```
The `Li` component was just created to inject into the markdown parser/render. We can just put this inline below much like how `ul` is handled. I would test this without `display: list-item` because that might have been a specific issue for that page.

avrohomgottlieb replied:
The following is what it looks like when removing the `display: list-item`. Looks like we lose the bullets.

<img width="717" alt="image" src="https://github.com/user-attachments/assets/4e3502de-42e9-4b85-902d-603442a5624f" />

davidsmejia commented on client/src/components/DatasetSamplesProjectOptions.js:
```diff
@@ -0,0 +1,20 @@
+import { Box, CheckBox } from 'grommet'
+import { FormField } from 'components/FormField'
+
+export const DatasetSampesProjectOptions = () => {
```
```suggestion
export const DatasetSamplesProjectOptions = () => {
```

davidsmejia commented on client/src/components/DatasetAddSamplesModal.js:
```diff
@@ -0,0 +1,84 @@
+                pad={{ left: '26px' }}
+                style={{ listStyle: 'disc' }}
+              >
+                <Li text="8 samples with single-cell modality" />
```
Can we extract values like this number as a variable to the top of this file so it will be easier to see what data we need when we go to integration?

avrohomgottlieb commented:
@davidsmejia I addressed all of your above feedback. Would like some clarity on @nozomione's question from above, and then we're ready to rock after Deepa gives it a glance 🎸 😎 .


> Nozomi's question:
> We reused the component defined for `DatasetAddProjectModal`. Since we'll have separate modals with separate contexts (i.e., project and sample), should we create a dedicated component for data format in each modal, or rename it to `DatasetDataFormat` to make it more generic? @davidsmejia

davidsmejia commented on client/src/components/DatasetAddSamplesModal.js:
```diff
@@ -0,0 +1,84 @@
+            </Heading>
+            <Box pad={{ top: 'small' }}>
+              <Box gap="medium" pad={{ bottom: 'medium' }} width="680px">
+                <DatasetProjectDataFormat />
```
This is right because this is the data format for the dataset and project so it is ok to keep the naming.

davidsmejia commented:
> @davidsmejia I addressed all of your above feedback. Would like some clarity on @nozomione's question from above, and then we're ready to rock after Deepa gives it a glance 🎸 😎 .
> 
> > Nozomi's question:
> > We reused the component defined for `DatasetAddProjectModal`. Since we'll have separate modals with separate contexts (i.e., project and sample), should we create a dedicated component for data format in each modal, or rename it to `DatasetDataFormat` to make it more generic? @davidsmejia

Let's keep the reuse and current name for now. The data format will derived from the project and not the sample.

davidsmejia reviewed (approved):
LGTM 🚀

nozomione reviewed (approved):
LGTM 🎉
